File: app/src/main/java/com/castelijns/mmdemo/user_detail/UserDetailPresenter.java
package com.castelijns.mmdemo.user_detail;

import com.castelijns.mmdemo.models.User;

public class UserDetailPresenter implements UserDetailContract.Presenter {

    private UserDetailContract.View view;
    private User user;

    UserDetailPresenter(UserDetailContract.View view, User user) {
        this.view = view;
        this.user = user;
    }

    @Override
    public void start() {

    }

    @Override
    public void stop() {

    }

    @Override
    public void onDirectionsClicked() {
        view.showNavigationTo(user.getAddress().getGeo().get("lat"),
                user.getAddress().getGeo().get("lon"));
    }
}

File: app/src/main/kotlin/com/castelijns/mmdemo/user_detail/UserDetailPresenter.kt
package com.castelijns.mmdemo.user_detail

import com.castelijns.mmdemo.models.User

class UserDetailPresenter internal constructor(private val view: UserDetailContract.View, private val user: User) : UserDetailContract.Presenter {

    override fun start() {

    }

    override fun stop() {

    }

    override fun onDirectionsClicked() {
        view.showNavigationTo(user.address.geo["lat"]!!, user.address.geo["lon"]!!)
    }
}
